code/game/objects/structures/window.dm

```diff
@@ -180,50 +180,65 @@
+	return TRUE
 
-	if(!(flags_1&NODECONSTRUCT_1) && !(reinf && state >= RWINDOW_FRAME_BOLTED))
-		if(I.tool_behaviour == TOOL_SCREWDRIVER)
-			to_chat(user, span_notice("You begin to [anchored ? "unscrew the window from":"screw the window to"] the floor..."))
-			if(I.use_tool(src, user, decon_speed, volume = 75, extra_checks = CALLBACK(src, .proc/check_anchored, anchored)))
-				set_anchored(!anchored)
-				to_chat(user, span_notice("You [anchored ? "fasten the window to":"unfasten the window from"] the floor."))
-			return
-		else if(I.tool_behaviour == TOOL_WRENCH && !anchored)
-			to_chat(user, span_notice("You begin to disassemble [src]..."))
-			if(I.use_tool(src, user, decon_speed, volume = 75, extra_checks = CALLBACK(src, .proc/check_state_and_anchored, state, anchored)))
-				var/obj/item/stack/sheet/G = new glass_type(user.loc, glass_amount)
-				if (!QDELETED(G))
-					G.add_fingerprint(user)
-				playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
-				to_chat(user, span_notice("You successfully disassemble [src]."))
-				qdel(src)
-			return
-		else if(I.tool_behaviour == TOOL_CROWBAR && reinf && (state == WINDOW_OUT_OF_FRAME) && anchored)
-			to_chat(user, span_notice("You begin to lever the window into the frame..."))
-			if(I.use_tool(src, user, 100, volume = 75, extra_checks = CALLBACK(src, .proc/check_state_and_anchored, state, anchored)))
-				state = RWINDOW_SECURE
-				to_chat(user, span_notice("You pry the window into the frame."))
-			return
+/obj/structure/window/screwdriver_act(mob/living/user, obj/item/tool)
+	if(!(flags_1 & NODECONSTRUCT_1) && !(reinf && state >= RWINDOW_FRAME_BOLTED))
+		to_chat(user, span_notice("You begin to [anchored ? "unscrew the window from":"screw the window to"] the floor..."))
+		if(tool.use_tool(src, user, decon_speed, volume = 75, extra_checks = CALLBACK(src, .proc/check_anchored, anchored)))
+			set_anchored(!anchored)
+			to_chat(user, span_notice("You [anchored ? "fasten the window to":"unfasten the window from"] the floor."))
+		return TRUE
+
+/obj/structure/window/wrench_act(mob/living/user, obj/item/tool)
+	if((flags_1 & NODECONSTRUCT_1) || (reinf && state >= RWINDOW_FRAME_BOLTED))
+		return FALSE
+	if(anchored)
+		return FALSE
+	to_chat(user, span_notice("You begin to disassemble [src]..."))
+	if(tool.use_tool(src, user, decon_speed, volume = 75, extra_checks = CALLBACK(src, .proc/check_state_and_anchored, state, anchored)))
+		var/obj/item/stack/sheet/G = new glass_type(user.loc, glass_amount)
+		if (!QDELETED(G))
+			G.add_fingerprint(user)
+		playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
+		to_chat(user, span_notice("You successfully disassemble [src]."))
+		qdel(src)
+	return TRUE
+
+/obj/structure/window/crowbar_act(mob/living/user, obj/item/tool)
+	if((flags_1 & NODECONSTRUCT_1) || (reinf && state >= RWINDOW_FRAME_BOLTED))
+		return FALSE
+	if(!(reinf && (state == WINDOW_OUT_OF_FRAME) && anchored))
```

Comment:
why check state twice? don't we only want to precede if it's `WINDOW_OUT_OF_FRAME`?
also I'd prefer if you didn't ! subblocks like this, makes things harder to read